src/core/common/CustomLayout.js

```diff
@@ -0,0 +1,97 @@
+import React, { Component } from 'react';
```

Comment:
You'd better decorate react-admin's `Layout`, overriding the `appBar` prop. As for the "isMenuVisible" logic, as I said in my previous comment it feels weird from a UI point of view.

Reply:
@fzaninotto 

> About the UI, it feels strange to me to have the patient summary in the top bar, above the menu, when the menu contains patient-independent items like "charts" and "patients".

Order of menu items was suggested by @tony-shannon :
"Charts on top, then Patients, then Patient Summary, then Problems, Medications, Vaccinations, Allergies, Contacts, Top Three Thing"

> You'd better decorate react-admin's Layout, overriding the appBar prop. 

I tried to do this and saved my results to this branch:
https://github.com/BogdanScherban/PulseTile-RA/tree/new-layout

But I don't find a way how to hide sidebar menu totally. For this reason:
1) In hide position we can see small part of sidebar with "..." (or Icon);
2) Empty space at the page where Sidebar should be absent (Charts and Patients list).

Problem is described on the video:
https://drive.google.com/file/d/1K-mFye-KDFL1ibf-lEZR7qkxXGjdYoPf/view?usp=sharing

Could you please tell me - is it possible to hide Sidebar area totally?

Reply:
> Order of menu items was suggested by @tony-shannon :
"Charts on top, then Patients, then Patient Summary, then Problems, Medications, Vaccinations, Allergies, Contacts, Top Three Thing"

Then I'm challenging @tony-shannon ;) HAving menus on the same level that depend or not on each other feels confusing to me.

> Could you please tell me - is it possible to hide Sidebar area totally?

Yes: instead of passing a custom `menu`, pass a custom `sidebar`. Handle your own show / hide logic inside that custom component.

Reply:
I used custom sidebar instead of custom menu. It works!!!